src/test-utils/smoke-test-utils.ts

```diff
@@ -96,10 +141,58 @@ export const isEachEventIngested = async (
             break;
         }
     }
+    return ingestedEvents;
+};
 
+/** Returns true when expected ingested event is found */
+export const expectEvents = async (
+    ingestedEvents: Map<string, string[]>,
+    eventIds: string[]
+) => {
     eventIds.forEach((eventId) => {
         if (!ingestedEvents.has(eventId)) {
             throw new Error(`Event ${eventId} not ingested.`);
         }
     });
+    return true;
+};
+
+/** Returns true when expected custom attributes are found */
+export const expectValidAttributes = async (
+    ingestedEvents: Map<string, string[]>,
+    eventIds: string[],
+    metadataAttributes: string[] | undefined = undefined
+) => {
+    if (metadataAttributes) {
+        eventIds.forEach((eventId) => {
+            expect(ingestedEvents.get(eventId)).toEqual(
+                expect.arrayContaining(metadataAttributes)
+            );
+        });
+    }
+    return true;
+};
+
+/** Returns true when no invalid custom attributes are found */
+export const expectInvalidAttributes = async (
+    ingestedEvents: Map<string, string[]>,
+    eventIds: string[],
+    metadataAttributes: string[] | undefined = undefined
+) => {
+    if (metadataAttributes) {
+        eventIds.forEach((eventId) => {
```

Comment:
question: Do we need this iteration? If the ingestedEvents has events with invalid metadata, it will always be there without having to go through each event. It will make this a lot simpler

Reply:
No, technically only need to check metadata of one event. Good catch! Will remove for loop here and above when checking for valid attributes.

Reply:
What I meant was you could just do filtering on `ingestedEvents` without having to do extra for loop on `eventIds` array :P